AI #2141: Apply changes to remove normative rules from editorial guidelines (#2191)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: guidelines/contributors/normative-requirements-guidelines.md

Reviewer: **Further adjustments to Normative Requirements Guidelines**

The following adjustments address remaining redundancies and consolidation opportunities identified during review:

- Remove the "Structuring Individual Dataset Requirements" and "Structuring Individual Column Requirements" sections entirely. Their content is either already covered elsewhere (asterisk rule in Editorial Style Guidelines, subject-first rule in the recommended pattern under "Normative Requirement Structure") or should be consolidated into Core sections as described below.
- Broaden the "Terminology Usage in Normative Requirements" subsection (currently under "Normative Subject") to cover all FOCUS entity references, not just columns. Entity references in normative requirements should use the EntityId (e.g., ColumnId, DatasetId, AttributeId), not Display Names.
- Move the "Default to Singular Form" guidance (currently in "Structuring Individual Column Requirements") into the appropriate Core section (e.g., "Terminology Usage in Normative Requirements") and generalize it to apply to all FOCUS entity references, not just columns.
- Remove the repeated "Important Note" about RECOMMENDED deprecation and the BCP 14 reference from all three entity-specific ordering sections ("Ordering of Dataset Requirements Within Groups", "Ordering of Column Requirements Within Groups", "Ordering of Attribute Requirements within Groups"). These are already covered in the Core section "Use of BCP 14 Keywords".

@joaquin and @ijurica  review and decide whether to address these in this PR or defer to a follow-up.

Author: I agreed with @ijurica that she can add these extra requirements in this PR.

<Subject (+qualifier)> + <BCP 14 Keyword> + <Verifiable State Descriptor> + <Object (+qualifier)> [+ Conditions]
```

> * Normative requirements MUST be expressed as individual bullet points.

Reviewer: ```suggestion
* Each normative requirement MUST be expressed as an individual bullet point, except for structural anchor requirements (see [Structural Anchor Requirement](#structural-anchor-requirement)).
```

* Each bullet MUST represent exactly one normative requirement expressing a single constraint.
* Each normative requirement MUST:
  * identify exactly one **normative subject** to which the requirement applies
  * contain exactly one **BCP 14 keyword** (MUST, SHOULD, MAY, MUST NOT, etc.), indicating the obligation level
  * express exactly one **verifiable constraint**

>   * be split into multiple bullets if it introduces multiple independent constraints.

Reviewer: ```suggestion
```

- Consider removing this from the "Each normative requirement MUST" composite bullet. This is a logical consequence of "express exactly one verifiable constraint" and is already covered in detail by the dedicated **Splitting Requirements** section.

Author: This requirement is already available in the Splitting Requirements section.

* Each normative requirement SHOULD describe a **verifiable state** of the object rather than behavior

### Explicit Conditions in Normative Requirements

* Normative requirements MUST include an explicit condition unless the rule applies universally.
* Conditional clauses MUST use standardized patterns such as:
   * when <condition>
   * if <condition>
   * unless <condition>  
   * only when <condition>  
   * only if <condition>  
   * except when <condition>  

>    * except if <condition>

Reviewer: consider...

```suggestion
* Conditional logic MUST be expressed using one of the following approved conditional keywords:
  * `when`
  * `unless`
  * `only when`
  * `except when`
```
**Question:** Should `when` be mandated over `if`, or at least defined as the preferred conditional keyword
Note: In earlier discussions (v1.2), we agreed to prefer `when` where both could be used.

Author: Agreed to keep "when" and removing "if"

* Each normative bullet MUST contain exactly one BCP 14 keyword (MUST, SHOULD, MAY, MUST NOT, SHOULD NOT). See [BCP14](https://tools.ietf.org/html/bcp14) [[RFC2119](https://tools.ietf.org/html/rfc2119)][[RFC8174](https://tools.ietf.org/html/rfc8174)].
* A bullet containing more than one normative keyword MUST be split.
* The term **RECOMMENDED** MUST NOT be used as a normative keyword. The keyword **SHOULD** MUST be used instead.

> * The lowercase word “recommended” MAY be used in non-normative text, provided it does not express a normative requirement.

Reviewer: ```suggestion
```

- Consider moving this to Editorial Style Guidelines. This governs non-normative text usage rather than normative requirement authoring.

Author: We agreed to drop these requirements as nothing stops an author from using "recommended".

* contains more than one BCP 14 keyword,

> * combines multiple obligations (e.g., multiple verifiable state descriptors, multiple objects, or multiple conditions that result in distinct constraints),

Author: incorporate "and", "or" as an example to this statement, then we can remove the new statement:
* combines multiple independent conditions using “and” or “or” that result in distinct constraints
